clis/tcli/src/bin/dtcli/daemon.rs

```diff
@@ -0,0 +1,119 @@
+
+use std::path::PathBuf;
+
+use anyhow::{self, Result};
+use sysinfo::{ProcessRefreshKind, RefreshKind, System};
+
+use crate::DTCLI_WD_PATH;
+
+/// Enum for figuring out the current process after the daemon is forked away
+/// from the API by the daemonization.
+pub enum ProcessType {
+    /// Getting this value after the daemonization means that the current process is the API.
+    Api,
+    /// Getting this value after the daemonization means that the current process is the Daemon.
+    Daemon,
+}
+
+/// Struct for packaging together the functions for managing the daemon.
+/// Currently does not need any members, since most of the data is stored
+/// in files anyway.
+pub struct Daemon;
+
+impl Daemon {
+    /// Checks whether the daemon is currently running.
+    /// Won't work if pid file had been tampered with or it's location changed.
+    ///
+    /// # Returns
+    ///
+    /// Result containing a bool. True means that the daemon is running.
+    pub fn is_running() -> Result<bool> {
+        let system = System::new_with_specifics(
+            RefreshKind::new().with_processes(ProcessRefreshKind::everything()),
+        );
+
+        match std::fs::read_to_string(get_wd_path()?.join("dtcli.pid")) {
+            Ok(pid_str) => {
+                let pid: usize = pid_str.trim().parse()?;
+                match system.process(sysinfo::Pid::from(pid)) {
+                    Some(_) => Ok(true),
+                    None => Ok(false),
+                }
+            }
+            Err(e) => match e.kind() {
+                std::io::ErrorKind::NotFound => Ok(false),
+                _ => Err(e.into()),
+            },
+        }
+    }
+
+    /// Function for conveniently printing the stderr file of the daemon.
+    /// Used by the API to quickly inform the user if something is wrong
+    /// with the daemon.
+    ///
+    /// # Returns
+    ///
+    /// The whole contents of the daemons stderr file as a String.
+    pub fn get_stderr() -> Result<String> {
+        let stderr_file = get_wd_path()?.join("dtcli.err");
+
+        let stderr_string = std::fs::read_to_string(stderr_file)?;
+
+        Ok(stderr_string)
+    }
+
+    /// Forks the daemon from the API process.
+    ///
+    /// # Returns
+    ///
+    /// Current process type indicating whether it's the API or the Daemon.
+    /// This is useful because of how the fork syscall.
+    pub fn start() -> Result<ProcessType> {
+        #[cfg(unix)]
+        {
+            use std::fs::File;
+            // This crate only works on unix systems and not Windows.
+            use daemonize::Daemonize;
+
+            let dtcli_wd_path = get_wd_path()?;
+
+            std::fs::create_dir_all(&dtcli_wd_path)?;
+            let stdout = File::create(dtcli_wd_path.join("dtcli.out"))?;
+            let stderr = File::create(dtcli_wd_path.join("dtcli.err"))?;
+
+            // TODO Check if nordvpn group exists and inform the user that it must be created, if it doesn't.
```

Comment:
nit: Updatw TODO to create the nordvpn group if it doesn't exist, instead of just informing

Reply:
Done